client/src/components/JurisdictionAdmin/BatchInventory.tsx

```diff
@@ -0,0 +1,502 @@
+  const areFilesUploaded =
+    isUploaded(cvrUpload) && isUploaded(tabulatorStatusUpload)
+  const isSignedOff = signOffQueries.status.data?.signedOffAt !== null
+  const initialStep = !areFilesUploaded
+    ? 'Upload Election Results'
+    : !isSignedOff
+    ? 'Inventory Batches'
+    : 'Download Audit Files'
+
+  const [step, setStep] = useState<typeof STEPS[number]>(initialStep)
+
+  return (
+    <Wrapper>
+      <Inner withTopPadding flexDirection="column">
+        <HeadingRow>
+          <H2>Batch Inventory</H2>
+          <LinkButton
+            minimal
+            to={`/election/${electionId}/jurisdiction/${jurisdictionId}`}
+            icon="chevron-left"
+          >
+            Back to Audit Source Data
+          </LinkButton>
+        </HeadingRow>
+        <StepContainer>
+          <StepProgress steps={STEPS} currentStep={step} />
+          {(() => {
+            switch (step) {
+              case 'Upload Election Results':
+                return (
+                  <UploadElectionResultsStep
+                    cvrUpload={cvrUpload}
+                    tabulatorStatusUpload={tabulatorStatusUpload}
+                    nextStep={() => setStep('Inventory Batches')}
+                  />
+                )
+              case 'Inventory Batches':
+                return (
+                  <InventoryBatchesStep
+                    worksheetUrl={`/election/${electionId}/jurisdiction/${jurisdictionId}/batch-inventory/worksheet`}
+                    signOffQueries={signOffQueries}
+                    prevStep={() => setStep('Upload Election Results')}
+                    nextStep={() => setStep('Download Audit Files')}
+                  />
+                )
+              case 'Download Audit Files':
+                return (
+                  <DownloadAuditFilesStep
+                    ballotManifestUrl={`/election/${electionId}/jurisdiction/${jurisdictionId}/batch-inventory/ballot-manifest`}
+                    batchTalliesUrl={`/election/${electionId}/jurisdiction/${jurisdictionId}/batch-inventory/batch-tallies`}
+                    prevStep={() => setStep('Inventory Batches')}
+                  />
+                )
+              default:
+                throw new Error('Unknown step')
+            }
+          })()}
+        </StepContainer>
+      </Inner>
+    </Wrapper>
+  )
+}
+
+const BatchInventory: React.FC = () => {
+  const { electionId, jurisdictionId } = useParams<{
+    electionId: string
+    jurisdictionId: string
+  }>()
+  const cvrUpload = useBatchInventoryCVR(electionId, jurisdictionId)
+  const tabulatorStatusUpload = useBatchInventoryTabulatorStatus(
+    electionId,
+    jurisdictionId
+  )
+  const signOffQueries = useBatchInventorySignOff(electionId, jurisdictionId)
+
+  if (
+    !(
+      cvrUpload.uploadedFile.isSuccess &&
+      tabulatorStatusUpload.uploadedFile.isSuccess &&
+      signOffQueries.status.isSuccess
+    )
+  ) {
+    return null
+  }
```

Comment:
Having a little bit of trouble understanding this condition - what state is it capturing? More specifically, under what circumstances can these `.isSuccess` booleans be false?